WIP: Run kafka, next eventing and last serving

matzew opened the pull request:
just for testing, revert the order ...

matzew commented:
infra

/retest

On Fri 16. Oct 2020 at 10:07, OpenShift Merge Robot <
notifications@github.com> wrote:

> @matzew <https://github.com/matzew>: The following test *failed*, say
> /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> ci/prow/4.5-upgrade-tests-aws-ocp-45 7cb2c7e
> <https://github.com/openshift-knative/serverless-operator/commit/7cb2c7e8ba0c97916cbd424e75ea4bd429e95195>
> link
> <https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/581/pull-ci-openshift-knative-serverless-operator-master-4.5-upgrade-tests-aws-ocp-45/1316977092734226432> /test
> 4.5-upgrade-tests-aws-ocp-45
>
> Full PR test history
> <https://prow.ci.openshift.org/pr-history?org=openshift-knative&repo=serverless-operator&pr=581>.
> Your PR dashboard
> <https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:matzew>.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-knative/serverless-operator/pull/581#issuecomment-709895106>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABGPTUHEIAIINQEV5H3EQ3SK75NRANCNFSM4SS4BP5Q>
> .
>
-- 
Sent from Gmail Mobile

matzew commented:
e2e was green, doing a new test:

/test 4.5-upstream-tests-aws-ocp-45

matzew commented:
/test 4.5-upstream-e2e-aws-ocp-45

matzew commented:
all green

/test 4.5-upstream-e2e-aws-ocp-45

matzew commented:
```
error: unable to recognize "https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.19.0/examples/kafka/kafka-persistent.yaml": no matches for kind "Kafka" in version "kafka.strimzi.io/v1beta1"
=============================================
==== WAITING FOR STRIMZI TO BECOME READY ====
=============================================
```

Strimzi CRD race ... should be fixed by them already - have not seen this in a log time


/retest

matzew commented:
Again, all green

/test 4.5-upstream-e2e-aws-ocp-45

aliok commented:
@matzew IMO this PR is the closest one to get merged.
I have some comments though.

aliok commented on hack/lib/serverless.bash:
```diff
@@ -4,7 +4,8 @@ function ensure_serverless_installed {
   if oc get knativeserving.operator.knative.dev knative-serving -n "${SERVING_NAMESPACE}" >/dev/null 2>&1 && \
-     oc get knativeeventing.operator.knative.dev knative-eventing -n "${EVENTING_NAMESPACE}" >/dev/null 2>&1
+     oc get knativeeventing.operator.knative.dev knative-eventing -n "${EVENTING_NAMESPACE}" >/dev/null 2>&1 && \
+     oc get knativekafka.operator.serverless.openshift.io knative-kafka -n "${EVENTING_NAMESPACE}" >/dev/null 2>&1
```
I just realized that this condition should only be considered if $INSTALL_KAFKA is true

aliok replied:
But, on the other hand, the install is idempotent, so, should be fine doing it twice in case serving and eventing is already installed and kafka isn't.

Again, I would change this as a follow up.

aliok replied:
PR against this PR's branch: https://github.com/matzew/serverless-operator/pull/3

aliok commented on knative-operator/pkg/apis/operator/v1alpha1/knativekafka_types.go:
```diff
@@ -69,3 +69,19 @@ type Channel struct {
+
+// Used in s-o e2e tests. can't instantiate there because vendor dirs
+// of s-o and s-o/knative-operator are different
+func NewKnativeKafka(name, namespace string, spec KnativeKafkaSpec) *KnativeKafka {
+	return &KnativeKafka{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KnativeKafka",
+			APIVersion: "operator.serverless.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: spec,
+	}
+}
```
Let's ignore this. Getting rid of this is an easy fix which can be done later.
Fix: https://github.com/openshift-knative/serverless-operator/pull/577/commits/370fc8530d15be95cdadf38c4f6ef4e5d88ae2d5

aliok replied:
PR against this PR's branch: https://github.com/matzew/serverless-operator/pull/3

aliok commented on README.md:
```diff
@@ -47,10 +47,15 @@ pushed to your docker repository.
 
 Use the appropriate make targets or scripts in `hack`:
 
+<<<<<<< HEAD
 - `make dev`: Deploys the serverless-operator without deploying Knative Serving and Eventing.
 - `make install`: Scales the cluster appropriately, deploys serverless-operator, Knative Serving and Eventing.
 - `make install-serving`: Scales the cluster appropriately, deploys serverless-operator and Knative Serving.
 - `make install-eventing`: Scales the cluster appropriately, deploys serverless-operator and Knative Eventing.
+=======
+- `make dev`: Deploys the serverless-operator without deploying Knative Serving, Eventing and Eventing-Contrib components.
+- `make install`: Scales the cluster appropriately, deploys serverless-operator, Knative Serving, Eventing and Eventing-Contrib components.
+>>>>>>> 7bf1c3f2... Enable Knative Kafka
```
I will fix push a commit to fix the conflict here

aliok commented on test/eventing-contrib.bash:
```diff
@@ -0,0 +1,67 @@
+  )
+}
+
+function run_e2e_tests_workaround(){
```
We can get rid of this workaround. This function was created because there were some broken/missing things in eventing-contrib/e2e-common.sh.

However, I would do this as a follow up. I wouldn't want to experiment if things have identical behavior when using the eventing-contrib one.

Let's leave as is.

aliok replied:
PR: https://github.com/matzew/serverless-operator/pull/5

matzew commented:
/retest

matzew commented:
Again, all green

/test 4.5-upstream-e2e-aws-ocp-45

matzew commented:
/test 4.5-upstream-e2e-aws-ocp-45

matzew commented:
`PullBuilderImageFailed` yay ....

aliok commented:
/test 4.5-upstream-e2e-aws-ocp-45

matzew commented:
missing here:
* CSV image override
* addressing review comments on this PR
* everything in knative-eventing namesments pace
* applying roles and clusterroles first, than deployetc
* Not deleting CRDs and CRs, when operator is deleting the `KnativeKafka`

matzew commented:
/refresh
/test all

matzew commented:
/refresh
/retest

matzew commented:
```
level=error
level=error msg="Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-5gycl6x6-85c73.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists]"
level=error msg="\tstatus code: 400, request id: fa0febb2-b0a1-4f6b-b4ca-581713ce596c"
level=error
level=error msg="  on ../tmp/openshift-install-278800715/route53/base.tf line 33, in resource \"aws_route53_record\" \"api_external\":"
level=error msg="  33: resource \"aws_route53_record\" \"api_external\" {"
level=error
level=error
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
error: failed to execute wrapped command: exit status 1

```

/retest

matzew commented:
/close
